spec/requests/companies_spec.rb

```diff
@@ -66,3 +66,15 @@
     expect(flash[:notice]).to eq('Companies list successfully updated')
   end
 end
+
+RSpec.describe 'GET /companies/?query=MO&commit=Search', type: :request do
+  it 'displays companies where param query is include in name or isin' do
+    Company.create!(name: 'MOSTALPLC', isin: 'PLMSTPL00018', condition: 'company active')
```

Comment:
Maybe you could create 2 companies and check that the second one is not displayed because it does not contain `MST` in its name.

Reply:
Yes, good idea, will add one more.